front/src/pages/teachers/teachers-information/index.js

```diff
@@ -0,0 +1,263 @@
+        <>
+          <Form {...layout} onFinish={this.handleNameChange}>
+            <Item
+              name="nameSurname"
+              align="center"
+              rules={[
+                { required: true, message: 'Please input new name and surname in English!' },
+                { validator: validateNameSurname },
+              ]}
+            >
+              <Input placeholder="Name in English" />
+            </Item>
+            <Form.Item align="center">
+              <SaveButton data-id={id} onClick={this.rememberId} htmlType="submit">
+                <SaveOutlined />
+              </SaveButton>
+            </Form.Item>
+          </Form>
+          <CancelButton data-id={id} onClick={this.cancelNameChanging}>
+            <RollbackOutlined />
+          </CancelButton>
+        </>
+      );
+    } else {
+      const { firstName, lastName } = this.props;
+      return (
+        <>
+          <NameSurname>{`${firstName} ${lastName}`}</NameSurname>
+          <EditButton data-id={id} onClick={this.changeName}>
+            <EditOutlined />
+          </EditButton>
+        </>
+      );
+    }
+  }
+
+  changeEmail(event) {
+    const id = +event.target.dataset.id;
+    const { changeEmail } = this.props;
+    changeEmail(id);
+  }
+
+  cancelEmailChanging(event) {
+    const id = +event.target.dataset.id;
+    const { cancelEmailChanging } = this.props;
+    cancelEmailChanging(id);
+  }
+
+  handleEmailChange(data) {
+    const { updateEmail, updatingId } = this.props;
+    data.id = updatingId;
+    updateEmail(data);
+  }
+
+  renderChangeEmailOrActualEmail(id, changeEmailIds) {
+    if (changeEmailIds.includes(id)) {
+      return (
+        <>
+          <Form {...layout} onFinish={this.handleEmailChange}>
+            <Item
+              name="email"
+              align="center"
+              rules={[
+                { required: true, message: 'Please input new email!' },
+                { type: 'email', message: 'The input is not valid E-mail!' },
+              ]}
+            >
+              <Input placeholder="Email" />
+            </Item>
+            <Form.Item align="center">
+              <SaveButton data-id={id} onClick={this.rememberId} htmlType="submit">
+                <SaveOutlined />
+              </SaveButton>
+            </Form.Item>
+          </Form>
+          <CancelButton data-id={id} onClick={this.cancelEmailChanging}>
+            <RollbackOutlined />
+          </CancelButton>
+        </>
+      );
+    } else {
+      const { email } = this.props;
+      return (
+        <>
+          <Email>{email}</Email>
+          <EditButton data-id={id} onClick={this.changeEmail}>
+            <EditOutlined />
+          </EditButton>
+        </>
+      );
+    }
+  }
+
+  handleDeleteTeacher() {
+    const { deleteTeacherId } = this.props;
+    this.props.deleteTeacher(deleteTeacherId)
+  }
+
+  showModalDelete(event) {
+    const id = +event.target.dataset.id;
+    this.id = id;
```

Comment:
Don't set any data on the component itself

Reply:
Removed